packer/rpc/communicator.go

```diff
@@ -137,14 +138,24 @@ func (c *communicator) UploadDir(dst string, src string, exclude []string) error
 func (c *communicator) Download(path string, w io.Writer) (err error) {
 	// Serve a single connection and a single copy
 	streamId := c.mux.NextId()
-	go serveSingleCopy("downloadWriter", c.mux, streamId, w, nil)
+
+	waitServer := make(chan bool)
```

Comment:
A nitpick but you should use a `chan struct{}` since that uses zero bytes. Then the easiest way to wait for it is to just `close()` it.